### PR TITLE
STREAMS-2699 integration with APIM

### PR DESCRIPTION
**Pull request by epiasentin-axway**
Thank you for your contribution to the Axway Streams documentation.

## Describe the changes

Describe how to secure Streams Hub and Subscribers APIs with APIM. 

https://deploy-preview-163--streams-open-docs.netlify.app/docs/install/apim-integration
https://deploy-preview-163--streams-open-docs.netlify.app/docs/install/customize-install/#activate-subscriber-sse-security

## Checklist for contributors

Before submitting this PR, please make sure:

* [x] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [x] You have signed the [Axway CLA](https://cla.axway.com/)
* [x] You have verified the technical accuracy of your change
* [x] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [x] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)

### CODE REVIEW

**Review by cremond — COMMENTED** [review]
Made a few comments. I think we also need to point to this page from that one https://deploy-preview-163--streams-open-docs.netlify.app/docs/install/amplify-central-integration/, at least once Romain's PR is merged (see his preview https://deploy-preview-164--streams-open-docs.netlify.app/docs/install/amplify-central-integration/)

File: content/en/docs/install/apim-integration.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: Integrate with APIM
```
Comment (andreamussap) [inline content/en/docs/install/apim-integration.md:2]: "Integrate Streams with API Management"

File: content/en/docs/install/apim-integration.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+Click on **Clients > Applications > New Application** button, fill the required information.
+
+* In **Application > API Access** section add All Streams Frontend APIs
+* In **Authentication > API Keys** section click on the **new API Key** button to generate a new API key
+
+You Streams installation is now secured by APIM.
+
+## Verify installation
+
+To access Streams APIs, an Authorization header must be provided. The format depends on the Security profile chosen during the Frontend creation:
```
[the remark this answers is not in the document]
Comment (epiasentin-axway) [reply content/en/docs/install/apim-integration.md:76]: agreed, I remove the OAuth part

File: content/en/docs/install/apim-integration.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+* For `OAuth` Security Profile
+
+  Use the **client_id** and **client_secret** created in your application
```
[the remark this answers is not in the document]
Comment (epiasentin-axway) [reply content/en/docs/install/apim-integration.md]: section removed

File: content/en/docs/install/apim-integration.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+If the configuration is correct, the returned HTTP code must be 2xx and the response body must contain the Topic created.
+
+### Validate Streams Subscribers SSE API
+
+Create a minimal SSE subscription with the previously created Streams Topic:
+
+```bash
+curl -v -X POST '${PUBLIC_API_GATEWAY_ADDRESS}/streams/subscribers/sse/api/v1/topics/testTopic/subscriptions' \
+-H 'Content-Type: application/json'  \
+-H '${APIM_AUTHORIZATION_HEADER}'  \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:125]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+```bash
+SUBSCRIPTION_ID="<ID_RETURNED>"
+```
+
+Two steps are required to consume a Streams SSE subscription:
+
+* Call the API Gateway to get a short-lived JWT token to access the Streams SSE Subscriber:
+
+    ```bash
+    curl -v -X GET '${PUBLIC_API_GATEWAY_ADDRESS}/streams/subscribers/sse/auth' \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:140]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    -H 'Content-Type: application/json' \
+    -H '${APIM_AUTHORIZATION_HEADER}'
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:142]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+    If the configuration is correct, the returned HTTP code must be 2xx and the JSON response must contain a field **token**. Use this field as SSE_JWT_TOKEN.
+
+    ```bash
+    SSE_JWT_TOKEN="<JWT_TOKEN_RETURNED>"
+    ```
+
+* Call directly the Streams SSE Subscriber with the token previously generated as `Authorization Bearer` header:
+
+    ```bash
+    curl -v -X GET '${PUBLIC_STREAMS_SUBSCRIBER_SSE_ADDRESS}/streams/subscribers/sse/api/v1/subscriptions/${SUBSCRIPTION_ID}/subscribe' \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:154]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+    -H 'Content-Type: application/json' \
+    -H 'Authorization: Bearer ${SSE_JWT_TOKEN}'
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:156]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    ```
+
+    If the configuration is correct, the returned HTTP code must be 2xx and the SSE stream should start receiving heartbeat events `:`.
+
+### Validate Streams Subscribers Webhook API
+
+Create a minimal Webhook subscription with the previously created Streams Topic. Set the mandatory field `webhookUrl` in the request body:
+
+```bash
+curl -v -X POST '${PUBLIC_API_GATEWAY_ADDRESS}/streams/subscribers/webhook/api/v1/topics/testTopic/subscriptions' \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:166]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+-H 'Content-Type: application/json'  \
+-H '${APIM_AUTHORIZATION_HEADER}'  \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:168]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/apim-integration.md [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+```
+
+If the configuration is correct, the returned HTTP code must be 2xx and the JSON response must contain a field **id**.
+
+### Validate Streams Subscribers Kafka API
+
+Create a minimal Kafka subscription with the previously created Streams Topic. Set mandatory fields `kafkaBootstrapServers` and `kafkaTopic` in the request body:
+
+```bash
+curl -v -X POST '${PUBLIC_API_GATEWAY_ADDRESS}/streams/subscribers/kafka/api/v1/topics/testTopic/subscriptions' \
```
Comment (rbissey-axway) [inline content/en/docs/install/apim-integration.md:181]: Variable will not be resolved -> the root cause is the usage of ' (simple quote).

File: content/en/docs/install/customize-install.md [hunk 12]
```diff
@@ -205,4 +205,30 @@ For example, to allow cross origin request from the domain name `https://origin-
...
+
+   ```bash
+   openssl genrsa -out key.pem 2048
+   openssl rsa -in key.pem -outform PEM -pubout -out cert.pem
+   ```
+
+* Create a kubernetes secret to store the RSA key pair
+
+   ```bash
+   kubectl create secret generic streams-subscriber-sse-jwt-secret --from-file=key=key.pem --from-file=cert=cert.pem -n ${NAMESPACE}
```
Comment (rbissey-axway) [inline content/en/docs/install/customize-install.md:224]: ${NAMESPACE} -> "${NAMESPACE}"

**Comment by epiasentin-axway** [comment]
@andreamussap could please have a look to this PR

**Pull request merged**